lib/iris/fileformats/grib/message.py

```diff
@@ -124,6 +160,15 @@ def data(self):
             raise TranslationError(fmt.format(template))
         return data
 
+    def __getstate__(self):
```

Comment:
> If this is not needed it should really go (and probably that pickle test too).
> 
>     ( Else we are just passing on the burden of not knowing why this is needed ! )

May I suggest, given the lack of feedback, that this is an issue that I can take on and follow up in a subsequent PR as required, rather than blocking this issue.

> or, if it is needed then this doesn't necessarily solve the problem, as an unpickled message could not access any keys not already read in.

I disagee a little, as the unpickled current implementation has the same feature, if the keys are not cached, then they are inaccessible following an unpickle

So, may I suggest that I create and follow up on the 'why is this here, is it needed' but we don't hold off on this PR to wait for that.  This is a blocking PR already, blocking a blocking PR awaiting more input makes me feel :(